Extensions/ArtifactEngine/Providers/filesystemProvider.ts

```diff
@@ -1,19 +1,22 @@
 import * as path from 'path';
 import * as fs from 'fs';
+var crypto = require('crypto');
 
 import * as models from '../Models';
 import { Logger } from '../Engine/logger';
 import { ArtifactItemStore } from '../Store/artifactItemStore';
 
-var tl = require('vsts-task-lib');
+var tl = require('vsts-task-lib/task');
 
 export class FilesystemProvider implements models.IArtifactProvider {
 
     public artifactItemStore: ArtifactItemStore;
 
-    constructor(rootLocation: string, rootItemPath?: string) {
+    constructor(rootLocation: string, rootItemPath?: string, cleanTargetDirectory?: boolean) {
```

Comment:
don't see a need to expose cleanTargetDirectory as an option. let's do it every time. we are anyways copying over from cache or source every time to target.